source/cellular_bg96_api.c

```diff
@@ -3195,6 +3338,187 @@ CellularError_t Cellular_GetHostByName( CellularHandle_t cellularHandle,
     return cellularStatus;
 }
 
+CellularError_t Cellular_DeleteFileFromStorage(CellularHandle_t cellularHandle, const char* filename)
+{
+    CellularContext_t* pContext = (CellularContext_t*)cellularHandle;
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+    char cmdBuf[CELLULAR_AT_CMD_TYPICAL_MAX_SIZE] = {'\0'};
+
+    CellularAtReq_t atReqFileDel = {
+        cmdBuf,
+        CELLULAR_AT_NO_RESULT,
+        NULL,
+        NULL,
+        NULL,
+        0,
+    };
+
+    if (cellularStatus == CELLULAR_SUCCESS)
+    {
+        /* Make sure the library is open. */
+        cellularStatus = _Cellular_CheckLibraryStatus(pContext);
+    }
+
+    if (cellularStatus == CELLULAR_SUCCESS)
+    {
+        /* Form the AT command. */
+
+        /* The return value of snprintf is not used.
+         * The max length of the string is fixed and checked offline. */
+        /* coverity[misra_c_2012_rule_21_6_violation]. */
+        (void)snprintf(cmdBuf, CELLULAR_AT_CMD_TYPICAL_MAX_SIZE, "%s%s", "AT+QFDEL=", filename);
+        pktStatus = _Cellular_TimeoutAtcmdRequestWithCallback(pContext, atReqFileDel, REQ_TIMEOUT_MS);
+
+        if (pktStatus != CELLULAR_PKT_STATUS_OK)
+        {
+            LogError(("Cellular_DeleteFileFromStorage: can't delete file, cmdBuf:%s, PktRet: %d", cmdBuf, pktStatus));
+            cellularStatus = _Cellular_TranslatePktStatus(pktStatus);
+        }
+    }
+
+    return cellularStatus;
+}
+
+CellularError_t Cellular_VerifyFileFromStorage(CellularHandle_t cellularHandle, const char* filename)
+{
+    CellularContext_t* pContext = (CellularContext_t*)cellularHandle;
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+    char cmdBuf[CELLULAR_AT_CMD_TYPICAL_MAX_SIZE] = {'\0'};
+
+    CellularAtReq_t atReqVerifyFile = {
+        cmdBuf,
+        CELLULAR_AT_NO_RESULT,
+        NULL,
+        NULL,
+        NULL,
+        0,
+    };
+
+    if (cellularStatus == CELLULAR_SUCCESS)
+    {
+        /* Make sure the library is open. */
+        cellularStatus = _Cellular_CheckLibraryStatus(pContext);
+    }
+
+    if (cellularStatus == CELLULAR_SUCCESS)
+    {
+        /* Form the AT command. */
+
+        /* The return value of snprintf is not used.
+         * The max length of the string is fixed and checked offline. */
+        /* coverity[misra_c_2012_rule_21_6_violation]. */
+        (void)snprintf(cmdBuf, CELLULAR_AT_CMD_TYPICAL_MAX_SIZE, "%s%s", "AT+QFLST=", filename);
+        pktStatus = _Cellular_TimeoutAtcmdRequestWithCallback(pContext, atReqVerifyFile, REQ_TIMEOUT_MS);

```

Comment:
Have you tested that this returns != CELLULAR_SUCCESS when the file doesn't exist? What's the expected behavior of this function when the file doesn't exist vs when it exists? I think we should have a parameter by reference in which we set its value depending on whether the file exists or not. Because the return status is reserved for error vs success.

`CellularError_t Cellular_CheckFileExistsInStorage(CellularHandle_t cellularHandle, const char* filename, uint8 *exists)`

Reply:
Removed